x/sync/network_client.go

```diff
@@ -248,7 +250,10 @@ func (c *networkClient) request(
 	}
 
 	handler := newResponseHandler()
-	c.outstandingRequestHandlers[requestID] = handler
+	c.outstandingRequestHandlers[requestID] = responseHandlerAndTime{
+		handler:     handler,
+		requestTime: time.Now(),
+	}
```

Comment:
Can't we measure the bandwidth only here? This function has all of the information needed right? It measures the start time here - and then it processes either a successful or failed request for the return value.

Feels like we don't need to introduce the `responseHandlerAndTime` struct or modify `getRequestHandler`.

Reply:
done